File: examples/blog/app/admin/middleware.go
package admin

import "github.com/emersonjoe/trilha"

// Middleware guards /admin: requires the session cookie set by /login.
func Middleware(c *trilha.Ctx, next trilha.Next) error {
	user, ok := c.Signed("sessao") // cookie assinado: não dá para forjar
	if !ok {
		return trilha.RedirectCode("/login?next="+c.Request().URL.Path, 302)
	}
	c.Set("user", user)
	return next()
}

File: testdata/next/middleware.ts
import { NextResponse } from "next/server"

export function middleware(request: Request) {
  const session = request.headers.get("cookie")
  if (!session) return NextResponse.redirect(new URL("/login", request.url))
  return NextResponse.next()
}

export const config = { matcher: ["/dashboard/:path*"] }
